pkg/cmd/cmd_run.go

```diff
@@ -35,60 +43,155 @@ func NewRunCmd() *cli.Command {
 			},
 		},
 		Action: func(c *cli.Context) error {
-			pwd, err := os.Getwd()
 
-			if err != nil {
-				reporter.ExitWithReport("kpm: internal bug: failed to load working directory")
+			tarPath := c.String("tar")
+			compileResult := ""
+
+			// If a tar package specified by "--tar" to run
+			if len(tarPath) != 0 {
+				absTarPath, err := absTarPath(tarPath)
+				if err != nil {
+					return err
+				}
+				// Extract the tar package to a directory with the same name.
+				// e.g.
+				// 'xxx/xxx/xxx/test.tar' will be extracted to the directory 'xxx/xxx/xxx/test'.
+				destDir := strings.TrimSuffix(absTarPath, filepath.Ext(absTarPath))
+				err = utils.UnTarDir(absTarPath, destDir)
+				if err != nil {
+					return err
+				}
+
+				// The directory after extracting the tar package is taken as the root directory of the package,
+				// and kclvm is called to compile the kcl program under the 'destDir'.
+				// e.g.
+				// if the tar path is 'xxx/xxx/xxx/test.tar',
+				// the 'xxx/xxx/xxx/test' will be taken as the root path of the kcl package to compile.
+				compileResult, compileErr := runPkgInPath(destDir, c.String("input"), c.Bool("vendor"))
+				// After compiling the kcl program, clean up the contents extracted from the tar package.
+				if utils.DirExists(destDir) {
+					err = os.RemoveAll(destDir)
+					if err != nil {
+						return err
+					}
+				}
+				if compileErr != nil {
+					return compileErr
+				}
+				fmt.Print(compileResult)
+			} else { // If no tar packages specified by "--tar" to run
+				// kpm will take the current directory ($PWD) as the root of the kcl package and compile.
+				pwd, err := os.Getwd()
+
+				if err != nil {
+					reporter.ExitWithReport("kpm: internal bug: failed to load working directory")
+				}
+
+				compileResult, err = runPkgInPath(pwd, c.String("input"), c.Bool("vendor"))
+				if err != nil {
+					return err
+				}
+				fmt.Print(compileResult)
 			}
+			return nil
+		},
+	}
+}
 
-			kclPkg, err := pkg.LoadKclPkg(pwd)
-			if err != nil {
-				reporter.ExitWithReport("kpm: failed to package pkg from ", pwd)
-				return err
-			}
+// runPkgInPath will load the 'KclPkg' from path 'pkgPath'.
+// And run the kcl package with entry file in 'entryFilePath' in 'vendorMode'.
+func runPkgInPath(pkgPath, entryFilePath string, vendorMode bool) (string, error) {
 
-			kclPkg.SetVendorMode(c.Bool("vendor"))
+	pkgPath, err := filepath.Abs(pkgPath)
+	if err != nil {
+		return "", errors.InternalBug
+	}
 
-			kpmHome, err := utils.GetAbsKpmHome()
-			if err != nil {
-				return err
-			}
+	kclPkg, err := pkg.LoadKclPkg(pkgPath)
+	if err != nil {
+		return "", errors.FailedToLoadPackage
```

Comment:
Why not use `return "", err`. or `return "", errors.Wrap(err)`?

Reply:
The number of code changes for this problem is relatively large for this PR, which will be adjusted later in the PR to fix https://github.com/KusionStack/kpm/issues/41 to optimize error message output.